ansible/roles/ssh-config/tasks/main.yaml

```diff
@@ -0,0 +1,12 @@
+- name: Disable password auth
+  ansible.builtin.lineinfile:
+    dest: /etc/ssh/sshd_config
+    regexp: '^#?\s*PasswordAuthentication\s'
+    line: 'PasswordAuthentication no'
+    state: present
```

Comment:
So this config does a regex-based find and replace for the entire line in `sshd_config`? From `*PasswordAuthentication*` to `PasswordAuthentication no`?

How do you prevent issues like the string `PasswordAuthentication` appearing in a comment or something and nuking that comment by accident?

Reply:
Welcome to Ansible land, where everything is chewing gum and shoestrings. Run while you still can.

You are 100% correct that [lineinfile](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/lineinfile_module.html#parameter-regexp) can be a foot-gun.

In this case it'll only replace the last instance, so no duplicates. `regexp` is for making the config look nice to a human (by picking the line a human would). In this case, picking the wrong line would not be as pretty, but would still work.